src/api.ts

```diff
@@ -39,8 +40,26 @@ class OpenAi {
   private openAI!: OpenAIApi;
 
   constructor() {
-    if (basePath) {
-      this.openAiApiConfiguration.basePath = basePath;
+    switch (apiType) {
+      case 'azure':
+        this.openAiApiConfiguration.baseOptions =  {
+          headers: {
+            "api-key": apiKey,
+          },
+          params: {
+            'api-version': '2023-03-15-preview',
```

Comment:
This needs to be configurable, depending on the specifics of the deployment. Another valid value is '2023-07-01-preview' and new ones will be added as time goes on.

Reply:
can we make this a configurable variable param?